### PR TITLE
Add features for Bluetooth APIs

### PR DESCRIPTION
**Pull request by dh79pyun**
### Change Description ###

Add Bluetooth features

### API Changes ###

 - ACR-897, ACR-1105, ACR-1106, ACR-1107

### CODE REVIEW

File: docs/application/native/tutorials/details/app-filtering.md [hunk 1]
```diff
@@ -201,7 +201,11 @@ package:
...
 | `http://tizen.org/feature/network.bluetooth.le` | Specify this key, if the application requires the Bluetooth LE feature. | 2.3   |
 | `http://tizen.org/feature/network.bluetooth.le.5_0` | Specify this key, if the application requires the Bluetooth 5 feature. | 5.0   |
+| `http://tizen.org/feature/network.bluetooth.le.ipsp` | Specify this key, if the application requires the Bluetooth Internet Protocol Support Profile (IPSP). | 4.0   |
+| `http://tizen.org/feature/network.bluetooth.le.gatt.client` | Specify this key, if the application requires the Bluetooth Generic Attribute Profile (GATT) Client. | 4.0   |
+| `http://tizen.org/feature/network.bluetooth.le.gatt.server` | Specify this key, if the application requires the Bluetooth Generic Attribute Profile (GATT) Server. | 4.0   |
 | `http://tizen.org/feature/network.bluetooth.opp` | Specify this key, if the application requires the Bluetooth Object Push feature (OPP). | 2.3   |
+| `http://tizen.org/feature/network.bluetooth.oob` | Specify this key, if the application requires the Bluetooth Out Of Band (OOB). | 4.0   |
```
Comment (divyk) [inline docs/application/native/tutorials/details/app-filtering.md]: Please make this small change:

Specify this key, if the application requires Bluetooth Out Of Band (OOB).

I removed 'the' as the sentence does not have a qualifier.

File: docs/application/native/tutorials/details/app-filtering.md [hunk 2]
```diff
@@ -380,7 +384,11 @@ Reference](../../api/overview.md).
...
```
Comment (divyk) [inline docs/application/native/tutorials/details/app-filtering.md]: Please make this small change:

Specify this key, if the application requires Bluetooth Out Of Band (OOB).

I removed 'the' as the sentence does not have a qualifier.

File: docs/application/native/guides/device/system.md [hunk 3]
```diff
@@ -409,7 +409,11 @@ The following table lists the network feature keys:
...
 | `http://tizen.org/feature/network.bluetooth.le.5_0` | `bool` | The platform returns `true` for this key, if the device supports Bluetooth 5 feature. |
+| `http://tizen.org/feature/network.bluetooth.le.ipsp` | `bool` | The platform returns `true` for this key, if the device supports Bluetooth Internet Protocol Support Profile (IPSP). |
+| `http://tizen.org/feature/network.bluetooth.le.gatt.client` | `bool` | The platform returns `true` for this key, if the device supports Bluetooth Generic Attribute Profile (GATT) Client. |
+| `http://tizen.org/feature/network.bluetooth.le.gatt.server` | `bool` | The platform returns `true` for this key, if the device supports Bluetooth Generic Attribute Profile (GATT) Server. |
 | `http://tizen.org/feature/network.bluetooth.opp` | `bool` | The platform returns `true` for this key, if the device supports Bluetooth Object Push Profile (OPP). |
+| `http://tizen.org/feature/network.bluetooth.oob` | `bool` | The platform returns `true` for this key, if the device supports Bluetooth Out Of Band (OOB)). |
```
[the remark this answers is not in the document]
Comment (dh79pyun) [reply docs/application/native/guides/device/system.md]: done

File: docs/application/native/guides/device/system.md [hunk 4]
```diff
@@ -409,7 +409,11 @@ The following table lists the network feature keys:
 | `http://tizen.org/feature/network.bluetooth.hid.device` | `bool` | The platform returns `true` for this key, if the device supports the Bluetooth Human Interface Device (HID) device feature. |
 | `http://tizen.org/feature/network.bluetooth.le` | `bool` | The platform returns `true` for this key, if the device supports Bluetooth Low Energy (BLE). |
 | `http://tizen.org/feature/network.bluetooth.le.5_0` | `bool` | The platform returns `true` for this key, if the device supports Bluetooth 5 feature. |
+| `http://tizen.org/feature/network.bluetooth.le.ipsp` | `bool` | The platform returns `true` for this key, if the device supports Bluetooth Internet Protocol Support Profile (IPSP). |
+| `http://tizen.org/feature/network.bluetooth.le.gatt.client` | `bool` | The platform returns `true` for this key, if the device supports Bluetooth Generic Attribute Profile (GATT) Client. |
```
[the remark this answers is not in the document]
Comment (dh79pyun) [reply docs/application/native/guides/device/system.md]: done

File: docs/application/native/guides/device/system.md [hunk 5]
```diff
@@ -409,7 +409,11 @@ The following table lists the network feature keys:
...
+| `http://tizen.org/feature/network.bluetooth.le.gatt.server` | `bool` | The platform returns `true` for this key, if the device supports Bluetooth Generic Attribute Profile (GATT) Server. |
```
[the remark this answers is not in the document]
Comment (dh79pyun) [reply docs/application/native/guides/device/system.md]: done

File: docs/application/native/tutorials/details/app-filtering.md [hunk 6]
```diff
@@ -201,7 +201,11 @@ package:
 | `http://tizen.org/feature/network.bluetooth.hid` | Specify this key, if the application requires the Bluetooth Human Input Device feature (HID). | 2.3   |
 | `http://tizen.org/feature/network.bluetooth.le` | Specify this key, if the application requires the Bluetooth LE feature. | 2.3   |
 | `http://tizen.org/feature/network.bluetooth.le.5_0` | Specify this key, if the application requires the Bluetooth 5 feature. | 5.0   |
+| `http://tizen.org/feature/network.bluetooth.le.ipsp` | Specify this key, if the application requires the Bluetooth Internet Protocol Support Profile (IPSP). | 4.0   |
+| `http://tizen.org/feature/network.bluetooth.le.gatt.client` | Specify this key, if the application requires the Bluetooth Generic Attribute Profile (GATT) Client. | 4.0   |
```
[the remark this answers is not in the document]
Comment (dh79pyun) [reply docs/application/native/tutorials/details/app-filtering.md]: done

File: docs/application/native/tutorials/details/app-filtering.md [hunk 7]
```diff
@@ -201,7 +201,11 @@ package:
...
+| `http://tizen.org/feature/network.bluetooth.le.gatt.server` | Specify this key, if the application requires the Bluetooth Generic Attribute Profile (GATT) Server. | 4.0   |
```
[the remark this answers is not in the document]
Comment (dh79pyun) [reply docs/application/native/tutorials/details/app-filtering.md]: done

File: docs/application/native/tutorials/details/app-filtering.md [hunk 8]
```diff
@@ -380,7 +384,11 @@ Reference](../../api/overview.md).
 | `http://tizen.org/feature/network.bluetooth.hid.device` | Specify this key, if the application requires the Bluetooth Human Interface Device (HID) device feature. | 3.0   |
 | `http://tizen.org/feature/network.bluetooth.le` | Specify this key, if the application requires the Bluetooth LE feature. | 2.3   |
 | `http://tizen.org/feature/network.bluetooth.le.5_0` | Specify this key, if the application requires the Bluetooth 5 feature. | 5.0   |
+| `http://tizen.org/feature/network.bluetooth.le.ipsp` | Specify this key, if the application requires the Bluetooth Internet Protocol Support Profile (IPSP). | 4.0   |
+| `http://tizen.org/feature/network.bluetooth.le.gatt.client` | Specify this key, if the application requires the Bluetooth Generic Attribute Profile (GATT) Client. | 4.0   |
```
[the remark this answers is not in the document]
Comment (dh79pyun) [reply docs/application/native/tutorials/details/app-filtering.md]: done

File: docs/application/native/tutorials/details/app-filtering.md [hunk 9]
```diff
@@ -380,7 +384,11 @@ Reference](../../api/overview.md).
...
+| `http://tizen.org/feature/network.bluetooth.le.gatt.server` | Specify this key, if the application requires the Bluetooth Generic Attribute Profile (GATT) Server. | 4.0   |
```
[the remark this answers is not in the document]
Comment (dh79pyun) [reply docs/application/native/tutorials/details/app-filtering.md]: done

File: docs/application/native/tutorials/details/app-filtering.md (hunk 1 above)
Comment (dh79pyun) [reply docs/application/native/tutorials/details/app-filtering.md]: done

File: docs/application/native/tutorials/details/app-filtering.md (hunk 2 above)
Comment (dh79pyun) [reply docs/application/native/tutorials/details/app-filtering.md]: done

Comment (divyk) [reply docs/application/native/tutorials/details/app-filtering.md]: Please refer again, I see the usage of 'the' in the preview. Please do the needed.

Comment (dh79pyun) [reply docs/application/native/tutorials/details/app-filtering.md]: done

**Comment by mijin85cho** [comment]
Suggestion.

How about following the alphabetical order for these added features?

Current : 
...
 - `http://tizen.org/feature/network.bluetooth.le.5_0`
 - `http://tizen.org/feature/network.bluetooth.le.ipsp`
 - `http://tizen.org/feature/network.bluetooth.le.gatt.client`
 - `http://tizen.org/feature/network.bluetooth.le.gatt.server`
 - `http://tizen.org/feature/network.bluetooth.opp`
 - `http://tizen.org/feature/network.bluetooth.oob`
 - `http://tizen.org/feature/network.bluetooth.phonebook.client`
...

Suggestion : 
...
 - `http://tizen.org/feature/network.bluetooth.le.5_0`
 - `http://tizen.org/feature/network.bluetooth.le.gatt.client`
 - `http://tizen.org/feature/network.bluetooth.le.gatt.server`
 - `http://tizen.org/feature/network.bluetooth.le.ipsp`
 - `http://tizen.org/feature/network.bluetooth.oob`
 - `http://tizen.org/feature/network.bluetooth.opp`
 - `http://tizen.org/feature/network.bluetooth.phonebook.client`
...

**Comment by mijin85cho** [comment]
> Suggestion.
> 
> How about following the alphabetical order for these added features?
> 
> Current :
> ...
> 
> * `http://tizen.org/feature/network.bluetooth.le.5_0`
> * `http://tizen.org/feature/network.bluetooth.le.ipsp`
> * `http://tizen.org/feature/network.bluetooth.le.gatt.client`
> * `http://tizen.org/feature/network.bluetooth.le.gatt.server`
> * `http://tizen.org/feature/network.bluetooth.opp`
> * `http://tizen.org/feature/network.bluetooth.oob`
> * `http://tizen.org/feature/network.bluetooth.phonebook.client`
>   ...
> 
> Suggestion :
> ...
> 
> * `http://tizen.org/feature/network.bluetooth.le.5_0`
> * `http://tizen.org/feature/network.bluetooth.le.gatt.client`
> * `http://tizen.org/feature/network.bluetooth.le.gatt.server`
> * `http://tizen.org/feature/network.bluetooth.le.ipsp`
> * `http://tizen.org/feature/network.bluetooth.oob`
> * `http://tizen.org/feature/network.bluetooth.opp`
> * `http://tizen.org/feature/network.bluetooth.phonebook.client`
>   ...

Hello, @dh79pyun 

I think this list order update would be the last thing to fix.

Would you please check the comment?

**Comment by dh79pyun** [comment]
> Suggestion.
> 
> How about following the alphabetical order for these added features?
> 
> Current :
> ...
> 
> * `http://tizen.org/feature/network.bluetooth.le.5_0`
> * `http://tizen.org/feature/network.bluetooth.le.ipsp`
> * `http://tizen.org/feature/network.bluetooth.le.gatt.client`
> * `http://tizen.org/feature/network.bluetooth.le.gatt.server`
> * `http://tizen.org/feature/network.bluetooth.opp`
> * `http://tizen.org/feature/network.bluetooth.oob`
> * `http://tizen.org/feature/network.bluetooth.phonebook.client`
>   ...
> 
> Suggestion :
> ...
> 
> * `http://tizen.org/feature/network.bluetooth.le.5_0`
> * `http://tizen.org/feature/network.bluetooth.le.gatt.client`
> * `http://tizen.org/feature/network.bluetooth.le.gatt.server`
> * `http://tizen.org/feature/network.bluetooth.le.ipsp`
> * `http://tizen.org/feature/network.bluetooth.oob`
> * `http://tizen.org/feature/network.bluetooth.opp`
> * `http://tizen.org/feature/network.bluetooth.phonebook.client`
>   ...

done

**Comment by dh79pyun** [comment]
> > Suggestion.
> > How about following the alphabetical order for these added features?
> > Current :
> > ...
> > 
> > * `http://tizen.org/feature/network.bluetooth.le.5_0`
> > * `http://tizen.org/feature/network.bluetooth.le.ipsp`
> > * `http://tizen.org/feature/network.bluetooth.le.gatt.client`
> > * `http://tizen.org/feature/network.bluetooth.le.gatt.server`
> > * `http://tizen.org/feature/network.bluetooth.opp`
> > * `http://tizen.org/feature/network.bluetooth.oob`
> > * `http://tizen.org/feature/network.bluetooth.phonebook.client`
> >   ...
> > 
> > Suggestion :
> > ...
> > 
> > * `http://tizen.org/feature/network.bluetooth.le.5_0`
> > * `http://tizen.org/feature/network.bluetooth.le.gatt.client`
> > * `http://tizen.org/feature/network.bluetooth.le.gatt.server`
> > * `http://tizen.org/feature/network.bluetooth.le.ipsp`
> > * `http://tizen.org/feature/network.bluetooth.oob`
> > * `http://tizen.org/feature/network.bluetooth.opp`
> > * `http://tizen.org/feature/network.bluetooth.phonebook.client`
> >   ...
> 
> Hello, @dh79pyun
> 
> I think this list order update would be the last thing to fix.
> 
> Would you please check the comment?

done

**Comment by dh79pyun** [comment]
> > Suggestion.
> > How about following the alphabetical order for these added features?
> > Current :
> > ...
> > 
> > * `http://tizen.org/feature/network.bluetooth.le.5_0`
> > * `http://tizen.org/feature/network.bluetooth.le.ipsp`
> > * `http://tizen.org/feature/network.bluetooth.le.gatt.client`
> > * `http://tizen.org/feature/network.bluetooth.le.gatt.server`
> > * `http://tizen.org/feature/network.bluetooth.opp`
> > * `http://tizen.org/feature/network.bluetooth.oob`
> > * `http://tizen.org/feature/network.bluetooth.phonebook.client`
> >   ...
> > 
> > Suggestion :
> > ...
> > 
> > * `http://tizen.org/feature/network.bluetooth.le.5_0`
> > * `http://tizen.org/feature/network.bluetooth.le.gatt.client`
> > * `http://tizen.org/feature/network.bluetooth.le.gatt.server`
> > * `http://tizen.org/feature/network.bluetooth.le.ipsp`
> > * `http://tizen.org/feature/network.bluetooth.oob`
> > * `http://tizen.org/feature/network.bluetooth.opp`
> > * `http://tizen.org/feature/network.bluetooth.phonebook.client`
> >   ...
> 
> Hello, @dh79pyun
> 
> I think this list order update would be the last thing to fix.
> 
> Would you please check the comment?

done

**Comment by dh79pyun** [comment]
Update the patchsets as the comments

File: docs/application/native/guides/device/system.md [hunk 10]
```diff
@@ -409,7 +409,11 @@ The following table lists the network feature keys:
...
 | `http://tizen.org/feature/network.bluetooth.le` | `bool` | The platform returns `true` for this key, if the device supports Bluetooth Low Energy (BLE). |
 | `http://tizen.org/feature/network.bluetooth.le.5_0` | `bool` | The platform returns `true` for this key, if the device supports Bluetooth 5 feature. |
+| `http://tizen.org/feature/network.bluetooth.le.gatt.client` | `bool` | The platform returns `true` for this key, if the device supports the Bluetooth Generic Attribute Profile (GATT) Client. |
+| `http://tizen.org/feature/network.bluetooth.le.gatt.server` | `bool` | The platform returns `true` for this key, if the device supports the Bluetooth Generic Attribute Profile (GATT) Server. |
+| `http://tizen.org/feature/network.bluetooth.le.ipsp` | `bool` | The platform returns `true` for this key, if the device supports Bluetooth Internet Protocol Support Profile (IPSP). |
 | `http://tizen.org/feature/network.bluetooth.opp` | `bool` | The platform returns `true` for this key, if the device supports Bluetooth Object Push Profile (OPP). |
```
Comment (mijin85cho) [inline docs/application/native/guides/device/system.md]: Question to this file and tutorials/details/app-filtering.md.

Shouldn't `http://tizen.org/feature/network.bluetooth.oob` come before `http://tizen.org/feature/network.bluetooth.opp`?

Comment (dh79pyun) [reply docs/application/native/guides/device/system.md]: done

**Pull request merged**